src/Monitor/Monitor/Alerts/AddAzureRmMetricAlertRuleV2Command.cs

```diff
@@ -0,0 +1,193 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Insights.OutputClasses;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Monitor;
+using Microsoft.Azure.Management.Monitor.Models;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Insights.Alerts
+{
+    /// <summary>
+    /// Add a GenV2 Metric Alert rule
+    /// </summary>
+    [Cmdlet("Add", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "MetricAlertRuleV2", SupportsShouldProcess = true), OutputType(typeof(PSMetricAlertRuleV2))]
```

Comment:
@AnanyShah this cmdlet implements `SupportsShouldProcess`, but it does not have a call to `ShouldProcess` in the executed code down below -- users providing the `-WhatIf` parameter, expecting it not to execute, will actually have the cmdlet execute